js/client-battle.js

```diff
@@ -396,6 +396,11 @@
 						if (move.id === 'Struggle' || move.id === 'Recharge') pp = '&ndash;';
 						if (move.id === 'Recharge') move.type = '&ndash;';
 						if (name.substr(0, 12) === 'Hidden Power') name = 'Hidden Power';
+						// Update Judgment's type to match that of the Plate type, if any.
+						var item = Tools.getItem(switchables[pos].item);
+						if (move.name === 'Judgment' && item.onPlate) {
+							move.type = item.onPlate;
```

Comment:
`move` is a pointer to global storage. Make a copy before messing with its type.